src/client/lcd/api/TxAPI.ts

```diff
@@ -446,7 +446,7 @@ export class TxAPI extends BaseAPI {
       })
       .then(d => SimulateResponse.fromData(d));
 
-    return new Dec(gasAdjustment).mul(simulateRes.gas_info.gas_used).toNumber();
+    return new Dec(gasAdjustment).mul(simulateRes.gas_info.gas_used).toDecimalPlaces(0, Dec.ROUND_UP).toNumber();
```

Comment:
gas_used is int64. So you can use number instead of Dec.

Reply:
The use of the `Dec` type is legacy code from the past.
However, `gasAdjustment` value can also be a `Decimal` type.
So to calculate all with the `number` type, the code is as follows:
```typescript
return Math.ceil(new Dec(gasAdjustment).toNumber() * simulateRes.gas_info.gas_used);
```
Additionally, because `gasAdjustment` may result in `estimateGas` being a real number with a decimal point when multiplied, I use `Math.ceil()` to round it up to an integer.
Is it ok?

Reply:
Since gasAdjustment doesn't need to support 18 decimal places, number is sufficient.